In [4]:
import pandas as pd
 
 
url = "https://raw.githubusercontent.com/mricardo89/data-mining/main/Unit-II/datasets/titanic.csv"
df = pd.read_csv(url)

print(df['Age'].skew(numeric_only=True))

0.38910778230082704


 1. ¿Qué significa físicamente sobre la edad de los pasajeros en el barco? (¿Había más jóvenes o más ancianos?)
skewness separa a las personas que tienen edad alta y edad baja dependiendo de un rango de numeros, habian mas jovenes que personas de mayor edad

In [5]:
import pandas as pd
 
 
url = "https://raw.githubusercontent.com/mricardo89/data-mining/main/Unit-II/datasets/titanic.csv"
df = pd.read_csv(url)

print(df['Fare'].kurt(numeric_only=True))

33.39814088089868


1. ¿Qué significa este número sobre el "grosor" de las colas de la distribución? ¿Confirma esto la existencia de los outliers que encontramos hoy en clase?
La columna Fare tiene una curtosis de 33.40, lo que indica una distribucion muy apuntada y con valores extremos. Esto significa que la mayoría de los pasajeros pagaron una tarifa baja, y otros una tarifa alta.
Definitivamente concuerda con los outliers vistos.


In [6]:
import pandas as pd
 
 
url = "https://raw.githubusercontent.com/mricardo89/data-mining/main/Unit-II/datasets/titanic.csv"
df = pd.read_csv(url)

print(pd.crosstab(df['Pclass'], df['Survived'], margins=True, normalize='index'))

Survived         0         1
Pclass                      
1         0.370370  0.629630
2         0.527174  0.472826
3         0.757637  0.242363
All       0.616162  0.383838


1. ¿Qué patrón logran visualizar rápidamente al cruzar ambas variables?
Los pasajeros de primera clase tuvieron una mayor proporcion de supervivencia que los de tercera clase, teniendo en cuenta que las clases superiores tenian la ventaja de ser rescatados antes que las clases bajas.
